static/app/views/performance/newTraceDetails/traceHeader/highlights.tsx

```diff
@@ -0,0 +1,133 @@
+import styled from '@emotion/styled';
+
+import {
+  EnvironmentHighlight,
+  HighlightsIconSummary,
+  ReleaseHighlight,
+} from 'sentry/components/events/highlights/highlightsIconSummary';
+import {space} from 'sentry/styles/space';
+import type {EventTransaction} from 'sentry/types/event';
+import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+import type {UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import {useTraceItemDetails} from 'sentry/views/explore/hooks/useTraceItemDetails';
+import {
+  OurLogKnownFieldKey,
+  type OurLogsResponseItem,
+} from 'sentry/views/explore/logs/types';
+import {TraceItemDataset} from 'sentry/views/explore/types';
+import {TraceHeaderComponents} from 'sentry/views/performance/newTraceDetails/traceHeader/styles';
+import {
+  TraceShape,
+  type TraceTree,
+} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
+
+function LogsHighlights({
+  logs,
+  organization,
+  project,
+}: {
+  logs: OurLogsResponseItem[];
+  organization: Organization;
+  project: Project | undefined;
+}) {
+  const log = logs[0];
+
+  const {data, isPending} = useTraceItemDetails({
+    traceItemId: String(log![OurLogKnownFieldKey.ID]),
+    projectId: String(log![OurLogKnownFieldKey.PROJECT_ID]),
+    traceId: String(log![OurLogKnownFieldKey.TRACE_ID]),
+    traceItemType: TraceItemDataset.LOGS,
+    referrer: 'api.explore.log-item-details', // TODO Abdullah Khan: Add new referrer for trace view header
+    enabled: !!logs,
+  });
+
+  if (!logs) {
```

Comment:
Are these boolean checks on `logs` necessary? Because `logs` is type `OurLogsResponseItem[]` which means `Boolean(logs)` is always `true`.

Reply:
We also do `const log = logs[0];` above, so even if `logs` could be `undefined` or `null`, this function would have errored at that line.

Reply:
Good point, I changed the prop of LogsHighlights from `logs: OurLogsResponseItem[]` to` log: OurLogsResponseItem` instead. And am passing it as such:

```
  if (tree.shape === TraceShape.EMPTY_TRACE && logs.length > 0) {
    return (
      <LogsHighlights
        log={logs[0] as OurLogsResponseItem}
        organization={organization}
        project={project}
      />
    );
  } 
```